apps/spruce/src/pages/image/PackagesTable/PackagesTable.test.tsx

```diff
@@ -0,0 +1,316 @@
+      expect(screen.queryAllByDataCy("packages-table-row")).toHaveLength(10);
+    });
+    await user.click(screen.getByTestId("lg-pagination-next-button"));
+    await waitFor(() => {
+      expect(screen.queryAllByDataCy("packages-table-row")).toHaveLength(5);
+    });
+  });
+});
+
+const imagePackagesPageOneMock: ApolloMock<
+  ImagePackagesQuery,
+  ImagePackagesQueryVariables
+> = {
+  request: {
+    query: IMAGE_PACKAGES,
+    variables: {
+      imageId: "ubuntu2204",
+      opts: {
+        page: 0,
+        limit: 10,
+      },
+    },
+  },
+  result: {
+    data: {
+      image: {
+        id: "ubuntu2204",
+        packages: {
+          data: [
+            {
+              manager: "pip 22.0.2 from (python 3.10)",
+              name: "alabaster",
+              version: "0.7.12",
+            },
+            {
+              manager: "pip 22.0.2 from (python 3.10)",
+              name: "attrs",
+              version: "21.2.0",
+            },
+            {
+              manager: "pip 22.0.2 from (python 3.10)",
+              name: "Automat",
+              version: "20.2.0",
+            },
+            {
+              manager: "pip 22.0.2 from (python 3.10)",
+              name: "Babel",
+              version: "2.8.0",
+            },
+            {
+              manager: "pip 22.0.2 from (python 3.10)",
+              name: "bcrypt",
+              version: "3.2.0",
+            },
+            {
+              manager: "pip 22.0.2 from (python 3.10)",
+              name: "beautifulsoup4",
+              version: "4.10.0",
+            },
+            {
+              manager: "pip 22.0.2 from (python 3.10)",
+              name: "blinker",
+              version: "1.4",
+            },
+            {
+              manager: "pip 22.0.2 from (python 3.10)",
+              name: "breezy",
+              version: "3.2.1",
+            },
+            {
+              manager: "pip 22.0.2 from (python 3.10)",
+              name: "certifi",
+              version: "2020.6.20",
+            },
+            {
+              manager: "pip 22.0.2 from (python 3.10)",
+              name: "chardet",
+              version: "4.0.0",
+            },
+          ],
+          filteredCount: 15,
+          totalCount: 15,
+        },
+      },
+    },
+  },
+};
+
+const imagePackagesNameFilterMock: ApolloMock<
+  ImagePackagesQuery,
+  ImagePackagesQueryVariables
+> = {
+  request: {
+    query: IMAGE_PACKAGES,
+    variables: {
+      imageId: "ubuntu2204",
+      opts: {
+        page: 0,
+        limit: 10,
+        name: "bcrypt",
+      },
+    },
+  },
+  result: {
+    data: {
+      image: {
+        id: "ubuntu2204",
+        packages: {
+          data: [
+            {
+              manager: "pip 22.0.2 from (python 3.10)",
+              name: "bcrypt",
+              version: "3.2.0",
+            },
+          ],
+          filteredCount: 1,
+          totalCount: 15,
+        },
+      },
```

Comment:
The reason why you're seeing GraphQL errors when you run the tests is because you need to include `__typename` in your mocks, like this:
```suggestion
 image: {
        __typename: "Image",
        id: "ubuntu2204",
        packages: {
           __typename: "ImagePackagesPayload",
          data: [
            {
              __typename: "Package",
              manager: "pip 22.0.2 from (python 3.10)",
              name: "bcrypt",
              version: "3.2.0",
            },
          ],
          filteredCount: 1,
          totalCount: 15,
        },
      },
```